test/raft/replication_test.cc

```diff
@@ -158,7 +158,19 @@ RAFT_TEST_CASE(take_snapshot, (test_case{
 // 2 nodes doing simple replication/snapshoting while leader's log size is limited
 RAFT_TEST_CASE(backpressure, (test_case{
          .nodes = 2,
-         .config = {{.snapshot_threshold = 10, .snapshot_trailing = 5, .max_log_size = 20}, {.snapshot_threshold = 20, .snapshot_trailing = 10}},
+         .config = {
+             {
+                 .snapshot_threshold = 10,
+                 .snapshot_trailing = 5,
+                 .max_snapshot_trailing_bytes = 150,
+                 .max_log_size = 450,
+                 .max_command_size = 150
```

Comment:
> Did you check that we still get backpressure?

Yes, this test and nemesis one hung on `semaphore::wait` when I initially missed the `memory_limit_approached` condition in `applier_fiber`.

We could have a `backpressure` metric with the rate of hitting the limit in `log_limiter_semaphore`. Then we could check in this test that this metric is not zero. This metric could also be useful on its own for diagnosis.

Reply:
I double-checked and you were right, the test didn't really do any backpressure, fixed that in a separate commit.

Gave up the backpressure metric, since it's hard to check the metrics of the concrete server, the metrics are static now.